File: code_samples/r/algorithms_string_algorithms.r
# Algorithms: String Algorithms
# AI/ML Training Sample

StringAlgorithms <- setRefClass(
    "StringAlgorithms",
    fields = list(data = "character"),
    methods = list(
        initialize = function() {
            data <<- ""
        },
        process = function(input) {
            data <<- input
        },
        getData = function() {
            return(data)
        },
        validate = function() {
            return(nchar(data) > 0)
        }
    )
)

# Example usage
instance <- StringAlgorithms$new()
instance$process("example")
cat("Data:", instance$getData(), "\n")
cat("Valid:", instance$validate(), "\n")

File: code_samples/dart/algorithms_string_algorithms.dart
// Algorithms: String Algorithms
// AI/ML Training Sample

class StringAlgorithms {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = StringAlgorithms();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}
